### PR TITLE
Add passt template

### PR DESCRIPTION
**Pull request by rhrazdil**
This PR adds kubevirt machine templates that use Passt network binding.
For that, kubevirt is bumped to v0.57.0 to have up-to-date API exposing the Passt binding.
Furthermore, Passt featureGate is enabled.

Signed-off-by: Radim Hrazdil <rhrazdil@redhat.com>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release notes**:

```release-note
Add template with Passt binding
Update kubevirt API to v0.57.0
```

### CODE REVIEW

File: templates/cluster-template-passt.yaml
Comment (davidvossel) [inline templates/cluster-template-passt.yaml:59]: is there any solution to the privileged port situation figured out yet? otherwise, we wouldn't be able to ssh into these guest nodes because port 22 won't be forwarded, right?

Comment (rhrazdil) [reply templates/cluster-template-passt.yaml:59]: That's correct, I have opened this PR: https://github.com/kubevirt/kubevirt/pull/8516
With that all looks good - ssh works, calico is ready

**Review by davidvossel — APPROVED** [review]
/lgtm
/approve

This won't 100% work until the kubevirt unpriv port pr lands for passt, but the template looks accurate.

**Comment by rhrazdil** [comment]
rebased

**Review by qinqon — CHANGES_REQUESTED** [review]
Can you remove the changes at "config" not related to this change ?

**Comment by qinqon** [comment]
> Can you remove the changes at "config" not related to this change ?

nah not needed

**Comment by qinqon** [comment]
/hold

@davidvossel do we wait for the kubevirt fix to be released to merge this ?

**Comment by davidvossel** [comment]
> @davidvossel do we wait for the kubevirt fix to be released to merge this ?

As long as we don't ship this template as part of the capk releases, i don't see an issue with merging it.

> Also we should test this at CI repeating the test on the same lane or doing a lane per favor.

I don't think that's necessary right now. Passt is more of an experimental feature at the moment for both capk and kubevirt. I'd consider this support best effort. We want to start experimenting with passt, and the template helps us with that.

**Comment by qinqon** [comment]
> > @davidvossel do we wait for the kubevirt fix to be released to merge this ?
> 
> As long as we don't ship this template as part of the capk releases, i don't see an issue with merging it.
> 
> > Also we should test this at CI repeating the test on the same lane or doing a lane per favor.
> 
> I don't think that's necessary right now. Passt is more of an experimental feature at the moment for both capk and kubevirt. I'd consider this support best effort. We want to start experimenting with passt, and the template helps us with that.

Is just another flavor that can be documented as experimental.

**Comment by rhrazdil** [comment]
[rebased](https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/compare/07a31d5bca55e83f121daacce11ed511449c2c96..6d6f3cafd3c95b526f693b1a44eabaa00d924e4b)

**Comment by qinqon** [comment]
Maybe flakiness is related to kubevirt bump https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/pull/183/files#diff-5aecb3b8f4bf3f07965c6a8cfec14fc7684939e8fddc46315b3d991e62e569b2R19

**Comment by rhrazdil** [comment]
Okay, to explain the issue above and the added [commit](https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/pull/183/commits/458b21eabb92662d554ea63d916c873c5a7614a1).
When rebooting a VM with persistent storage, we need to make sure that we have persistant MAC address - either configure static mac in the template or for more general approach, use KubeMacPool. That's most likely the approach we'd like to take.

See https://github.com/kubevirt/kubevirt/issues/6410 for more info.

File: templates/cluster-template-persistent-storage-kccm.yaml
Comment (davidvossel) [inline templates/cluster-template-persistent-storage-kccm.yaml:70]: I don't think we can set static mac addresses in the template because the same mac will end up being used for all VMs stamped from this template.

Comment (qinqon) [reply templates/cluster-template-persistent-storage-kccm.yaml:70]: This is only to exercise CI with fixed mac's, finally we should use kubemacpool.

**Comment by rhrazdil** [comment]
Updated the kubevirt release to v0.58.0, which contains the `passt` fix for lower ports

**Comment by qinqon** [comment]
/lgtm
Are we going to use kubemacpool so we don't have to harcode MACs ?

**Pull request merged**
